### PR TITLE
Stop re-pulling image on every container start

### PR DESCRIPTION
**Pull request by nzoueidi**
**What type of PR is this?**
cleanup
feature

**Which issue does this PR fix**: #1249 

**What does this PR do / Why do we need it**:
Change imagePullPolicy to IfNotPresent, to reduce the boot delay and the load on ECR.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by dallasmarlow** [comment]
I think this PR would make more sense if the docker image used had a specific version tag (IMHO).

**Comment by anguslees** [comment]
> I think this PR only makes sense for manifests using a specific version tag instead of `latest` (IMHO).

Agreed.  I note this is also exactly the Kubernetes default behaviour, so I suggest we just _remove_ `imagePullPolicy` entirely, rather than try to set it to either value.

File: config/master/manifests.jsonnet [hunk 1]
```diff
@@ -144,7 +144,7 @@ local awsnode = {
           containers_:: {
             awsnode: {
               image: "%s/amazon-k8s-cni:%s" % [$.ecrRepo, $.version],
-              imagePullPolicy: "Always",
+              imagePullPolicy: "IfNotPresent",
```
Comment (anguslees) [inline config/master/manifests.jsonnet:147]: Another option is to set this conditionally in jsonnet:
```jsonnet
imagePullPolicy: if $.version == "latest" then "Always" else "IfNotPresent",
```

Tbh, I've found that more trouble than it's worth in the past (eg: a naive kubectl patch might change the image, but forget to update the imagePullPolicy) so again I suggest just removing the field entirely and using the k8s dynamic default.

Comment (nzoueidi) [reply config/master/manifests.jsonnet:147]: yeah, you are totally right, the behavior of k8s is by default "IfNotPresent". So we proceed with removing the fields and just mention the condition in the jsonnet manifest. I will update the PR accordingly.

**Comment by nzoueidi** [comment]
I modified the changes and squashed my commits. Let me know what do you think.
cc @dallasmarlow  @anguslees

**Comment by dallasmarlow** [comment]
@nzoueidi it looks like your change may have introduced a spacing issue.

**Comment by nzoueidi** [comment]
It looks like the spaces are shown only in the PR of Github

**Comment by anguslees** [comment]
I think you have not regenerated the manifests following your latest jsonnet change.  Please re-run `make generate`.

Also, I think I may have confused you above - since you've followed _both_ of my alternate solutions :sweat_smile:.  To be clearer: I think we should remove the jsonnet if/else conditional, and not set `imagePullPolicy` _at all_.  So, can you:
0. (Rebase, since github says your branch has fallen behind)
1. Remove `imagePullPolicy: if $.version == "latest" then "Always" else "IfNotPresent"` line from jsonnet.
2. Re-run `make generate`.
3. Confirm the generated yaml now has _no_ `imagePullPolicy`.
4. Update this PR with all that.

Awesome, thanks in advance.  And sorry for the run-around.

**Comment by jayanthvn** [comment]
Hi @nzoueidi , 

Can you please make the changes suggested by @anguslees. Then I can merge the PR.

**Comment by nzoueidi** [comment]
sorry for the delay, will do today

**Comment by jayanthvn** [comment]
Thanks @nzoueidi

**Review by anguslees — APPROVED** [review]

**Comment by jayanthvn** [comment]
Hi @nzoueidi 

Can you please rebase your branch? Then I should be able to merge it.

**Comment by nzoueidi** [comment]
Thank you @jayanthvn, done.

**Pull request merged**